### PR TITLE
Fixes player megafauna not regaining any health from eating

### LINKED ISSUE #23435: lesser ash drake regen is broke
Gibbing someone regenerates your health, but the limbs hitting you takes it away and then makes you go down even more.
EDIT: Yes, this is a staff of change ash drake.

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Fixes #23435

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 1]
```diff
@@ -101,7 +101,7 @@
 	visible_message(
 		"<span class='danger'>[src] devours [L]!</span>",
 		"<span class='userdanger'>You feast on [L], restoring your health!</span>")
-	if(!(z in GLOB.station_z_levels && !client)) //NPC monsters won't heal while on station
+	if(!(z in GLOB.station_z_levels || client)) //NPC monsters won't heal while on station
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: these in checks aren't properly separated

**Comment by QualityVan** [comment]
Did you test this? I feel like the logic isn't doing what you want it to be doing.

**Comment by optimumtact** [comment]
the issue is the brackets, changes the || back to an &&

**Comment by KorPhaeron** [comment]
Players are supposed to be able to heal on station so having a client is an or to being off station

**Pull request merged**